hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapValuesWithPagingPredicateMessageTask.java

```diff
@@ -43,12 +43,12 @@ public MapValuesWithPagingPredicateMessageTask(ClientMessage clientMessage, Node
 
     @Override
     protected Object reduce(Collection<QueryResultRow> result) {
-        PagingPredicateImpl pagingPredicate = (PagingPredicateImpl) getPredicate();
-        List<Map.Entry<Data, Data>> entriesData = getSortedPageEntries(result, pagingPredicate);
+        List<Map.Entry<Data, Data>> entriesData = getSortedPageEntries(result);
```

Comment:
We are still creating two lists here (`entriesData` and `valueList`), which seems unnecessary. Can it be avoided?